### DISCUSSION TITLE
Getting errors despite Strongswan running

### DISCUSSION THREAD

**Discussion by BR41N14C** [strongswan/strongswan#1737] [Q&A]
Hello everyone,
I have a strongswan linux VM inside Azure (VNet and VGW).
I have set everything up which i can show if needed, but i keep getting these errors. I am not sure where else to look at. Would appreciate some guidance:

**Logs:**
```
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: unable to create IPv4 routing table rule
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: unable to create IPv6 routing table rule
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: loading ca certificates from '/etc/ipsec.d/cacerts'
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]:   loaded ca certificate "C=UK, O=Company_name, CN=Company_name Root CA" from '/etc/ipsec.d/cacerts/ca-cert.pem'
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: loading aa certificates from '/etc/ipsec.d/aacerts'
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: loading attribute certificates from '/etc/ipsec.d/acerts'
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: loading crls from '/etc/ipsec.d/crls'
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: loading secrets from '/etc/ipsec.secrets'
Jun 13 13:26:15 Company-Strongswan python3[769]:  -- Logs begin at Mon 2023-06-12 14:03:23 UTC, end at Tue 2023-06-13 13:26:15 UTC. --
Jun 13 13:26:15 Company-Strongswan python3[769]: Jun 13 13:26:07 Company-Strongswan python3[228]: Setting up firewall for the WALinux Agent with args: {'dst_ip': 'Public_IP', 'uid': '0>
Jun 13 13:26:15 Company-Strongswan python3[769]: Jun 13 13:26:07 Company-Strongswan python3[228]: Successfully set the firewall rules
Jun 13 13:26:15 Company-Strongswan python3[769]: Jun 13 13:26:07 Company-Strongswan systemd[1]: Finished Setup network rules for WALinuxAgent.
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]:   loaded RSA private key from '/etc/ipsec.d/private/host-vpn-key.pem'
Jun 13 13:26:15 Company-Strongswan charon: 00[LIB] loaded plugins: charon test-vectors ldap pkcs11 tpm aesni aes rc2 sha2 sha1 md5 mgf1 rdrand random nonce x509 revocation constraints pubkey >
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: loaded 0 RADIUS server configurations
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: HA config misses local/remote address
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: loaded plugins: charon-systemd test-vectors ldap pkcs11 tpm aesni aes rc2 sha2 sha1 md5 mgf1 rdrand random nonce x509 revocation constr>
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: dropped capabilities, running as uid 0, gid 0
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: spawning 16 worker threads
Jun 13 13:26:15 Company-Strongswan charon: 00[LIB] dropped capabilities, running as uid 0, gid 0
Jun 13 13:26:15 Company-Strongswan charon: 00[JOB] spawning 16 worker threads
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: received stroke: add connection 'azureTunnel'
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: adding virtual IP address pool IP_here
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]:   loaded certificate "C=UK, O=Company_name, CN=Azure_Gateway_PublicIP" from 'host-vpn-cert.pem'
Jun 13 13:26:15 Company-Strongswan charon-systemd[705]: added configuration 'azureTunnel'
Jun 13 13:26:15 Company-Strongswan swanctl[1094]: no files found matching '/etc/swanctl/conf.d/*.conf'
Jun 13 13:26:15 Company-Strongswan swanctl[1094]: no authorities found, 0 unloaded
Jun 13 13:26:15 Company-Strongswan swanctl[1094]: no pools found, 0 unloaded
Jun 13 13:26:15 Company-Strongswan swanctl[1094]: no connections found, 0 unloaded
Jun 13 13:26:15 Company-Strongswan systemd[1]: Started strongSwan IPsec IKEv1/IKEv2 daemon using swanctl.

```

This is what I get when i run status:
![image](https://github.com/strongswan/strongswan/assets/30053975/3d580e3f-6b60-4a5d-b471-edf67c1c3a3b)

**Comment by tobiasbrunner** [comment]
You are running two IKE daemons, make sure you deactivate one (see e.g. https://github.com/strongswan/strongswan/discussions/1440#discussioncomment-4456389 or https://github.com/strongswan/strongswan/discussions/1638#discussioncomment-5521207).

**Reply by BR41N14C** [reply]
thanks for your answer @tobiasbrunner 
I tried disabling it using different commands but i get an error:
I tried:
  655  sudo systemctl stop strongswanupdate-rc.d charon-systemd disable
  656  update-rc.d charon-systemd disable
  658  service charon-systemd stop
Failed to stop charon-systemd.service: Unit charon-systemd.service not loaded.
  659  sudo systemctl disable charon-systemd
Failed to disable unit: Unit file charon-systemd.service does not exist.
  660  sudo systemctl stop charon-systemd
Failed to stop charon-systemd.service: Unit charon-systemd.service not loaded.

**Reply by BR41N14C** [reply]
sudo systemctl disable strongswan.service
Removed /etc/systemd/system/strongswan-swanctl.service.
Removed /etc/systemd/system/multi-user.target.wants/strongswan.service.

Would this work? saw on Stack

**Reply by BR41N14C** [reply]
Sorry for the spam,  so I changed the network interface to static from dynamic and a connection is being picked up but its getting dropped:
![image](https://github.com/strongswan/strongswan/assets/30053975/c260bb43-df54-439c-a067-88558c9bd969)

**Reply by tobiasbrunner** [reply]
Read the logs.

**Reply by BR41N14C** [reply]
@tobiasbrunner I tried once but it was overwhelming, but at a closer inspection, I see that the connection is getting dropped:
`Jun 15 15:00:29 VPN-Strongswan charon: 09[ENC] generating IKE_AUTH response 5 [ EAP/REQ/TLS ]
Jun 15 15:00:29 VPN-Strongswan charon: 09[NET] sending packet: from VPN_IP[4500] to MY_PUBLIC_IP[4500] (176 bytes)
Jun 15 15:00:29 VPN-Strongswan charon: 10[NET] received packet: from MY_PUBLIC_IP[4500] to VPN_IP[4500] (96 bytes)
Jun 15 15:00:29 VPN-Strongswan charon: 10[ENC] parsed IKE_AUTH request 6 [ EAP/RES/TLS ]
Jun 15 15:00:29 VPN-Strongswan charon: 10[TLS] received TLS close notify
Jun 15 15:00:29 VPN-Strongswan charon: 10[TLS] sending TLS close notify`

**Reply by tobiasbrunner** [reply]
Yes, by the client. Maybe it doesn't like the server certificate.

**Reply by BR41N14C** [reply]
weird that, i feel im using a pretty generic template for the vpn cert:

Server:
`ipsec pki --pub --in private/host-vpn-key-publicip.pem --type rsa | ipsec pki --issue --lifetime 730 --cacert cacerts/ca-cert.pem  --cakey private/ca-key.pem--dn "C=$COUNTRY_CODE, O=$ORGANISATION, CN=$SERVER_NAME" --san=$SERVER_NAME --san $IP_ADDRESS  --san @$IP_ADDRESS --flag serverAuth --flag ikeIntermediate --outform pem > certs/host-vpn.pem`

Root:
` ipsec pki --self --ca --lifetime 3650 --in private/ca-key.pem --type rsa --dn "C=UK, O=Company, CN=Company Root CA" --outform pem > cacerts/ca-cert.pem`

User:
`ipsec pki --pub --in private/strongswan-key.pem --type rsa | ipsec pki --issue --lifetime 730 --cacert cacerts/ca-cert.pem --cakey private/ca-key.pem --dn "C=UK, O=Company, CN=username@gmail.com" --san "username@gmail.com" --outform pem > certs/strongswan-cert.pem`

**Reply by tobiasbrunner** [reply]
Maybe the CA cert is missing on the client. Or there could be an issue with the user cert/key. You have to check the log there for details.

**Reply by BR41N14C** [reply]
yeah I think (hope) the CA is done correctly, the logs on the client i am seeing is:
`2023-06-15 16:57:31.155552 (pid/13147 [NEIKEv2Provider]) <Notice>: Service stub created for com.apple.security.XPCAcmeService
2023-06-15 16:57:31.155733 (pid/13147/com.apple.MTLCompilerService) <Notice>: created the implicit endpoint upfront
2023-06-15 16:57:31.155741 (pid/13147/com.apple.MTLCompilerService) <Notice>: draining messages from com.apple.MTLCompilerService
2023-06-15 16:57:31.156594 (pid/12932/com.apple.NetworkExtension.IKEv2Provider [13147]) <Notice>: extension already interrupted clean exit attempt
2023-06-15 16:57:31.451801 (pid/12932/com.apple.NetworkExtension.IKEv2Provider [13147]) <Notice>: exited due to SIGKILL | sent by nesessionmanager[437]
2023-06-15 16:57:31.451807 (pid/12932/com.apple.NetworkExtension.IKEv2Provider [13147]) <Notice>: process exited in a dirty state
2023-06-15 16:57:31.451811 (pid/12932/com.apple.NetworkExtension.IKEv2Provider [13147]) <Notice>: service state: exited
2023-06-15 16:57:31.451832 (pid/12932/com.apple.NetworkExtension.IKEv2Provider [13147]) <Notice>: internal event: EXITED, code = 0
2023-06-15 16:57:31.451836 (pid/12932/com.apple.NetworkExtension.IKEv2Provider [13147]) <Notice>: job state = exited
2023-06-15 16:57:31.451861 (pid/12932 [neagent]) <Notice>: service inactive: com.apple.NetworkExtension.IKEv2Provider`

seems like something is killing the session but doesnt really say what

**Reply by BR41N14C** [reply]
checking the syslog a bit more i found this:

`Jun 15 16:18:50 CompanyVPN-Strongswan charon: 05[IKE] MY_LOCAL_IP is initiating an IKE_SA
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 05[IKE] local host is behind NAT, sending keep alives
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 05[IKE] remote host is behind NAT
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 05[IKE] sending cert request for "C=UK, O=Company, CN=Company Root CA"
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 05[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(CHDLESS_SUP) N(MULT_AUTH) ]
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 05[NET] sending packet: from VM_PRIVATE_IP[500] to MY_LOCAL_IP[500] (481 bytes)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[NET] received packet: from MY_LOCAL_IP[4500] to VM_PRIVATE_IP[4500] (352 bytes)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[ENC] unknown attribute type INTERNAL_DNS_DOMAIN
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[ENC] parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr CPRQ(ADDR MASK DHCP DNS ADDR6 DHCP6 DNS6 DOMAIN) N(ESP_TFC_PAD_N) N(NON_FIR>
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[IKE] initiating EAP_IDENTITY method (id 0x00)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[IKE] received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[IKE] peer supports MOBIKE
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[IKE] authentication of 'VM_PUBLIC_IP (myself) with RSA signature successful
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[IKE] sending end entity cert "C=UK, O=Company, CN=Company Root CA"
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[ENC] generating IKE_AUTH response 1 [ IDr CERT AUTH EAP/REQ/ID ]
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[ENC] splitting IKE message (1968 bytes) into 2 fragments
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[ENC] generating IKE_AUTH response 1 [ EF(1/2) ]
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[ENC] generating IKE_AUTH response 1 [ EF(2/2) ]
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[NET] sending packet: from VM_PRIVATE_IP[4500] to MY_LOCAL_IP[4500] (1236 bytes)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 06[NET] sending packet: from VM_PRIVATE_IP[4500] to MY_LOCAL_IP[4500] (804 bytes)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 08[NET] received packet: from MY_LOCAL_IP[4500] to VM_PRIVATE_IP[4500] (112 bytes)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 08[ENC] parsed IKE_AUTH request 2 [ EAP/RES/ID ]
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 08[IKE] received EAP identity 'username@Company.com'
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 08[IKE] initiating EAP_TLS method (id 0x49)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 08[ENC] generating IKE_AUTH response 2 [ EAP/REQ/TLS ]
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 08[NET] sending packet: from VM_PRIVATE_IP[4500] to MY_LOCAL_IP[4500] (80 bytes)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 09[NET] received packet: from MY_LOCAL_IP[4500] to VM_PRIVATE_IP[4500] (240 bytes)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 09[ENC] parsed IKE_AUTH request 3 [ EAP/RES/TLS ]
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 09[TLS] negotiated TLS 1.2 using suite TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 09[TLS] sending TLS server certificate 'C=UK, O=Company, CN=Company Root CA'
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 09[TLS] sending TLS cert request for 'C=UK, O=Company, CN=Company Root CA'
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 09[ENC] generating IKE_AUTH response 3 [ EAP/REQ/TLS ]
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 09[NET] sending packet: from VM_PRIVATE_IP[4500] to MY_LOCAL_IP[4500] (1104 bytes)
Jun 15 16:18:50 CompanyVPN-Strongswan charon: 07[NET] received packet: from MY_LOCAL_IP[4500] to VM_PRIVATE_IP[4500] (80 bytes)
`

**Reply by tobiasbrunner** [reply]
No other/more specific log messages on the client before the exit?

**Reply by BR41N14C** [reply]
@tobiasbrunner thats all i have

**Reply by tobiasbrunner** [reply]
Then there really isn't much we can do here. Other than the reasons already mentioned, I don't know why the client would terminate the connection.

**Reply by BR41N14C** [reply]
im with you regarding the server cert, it probably is that but i cant see what is going wrong in the creation, the errors im getting now is:
Jun 16 16:14:11 Strongswan charon: 07[IKE] local host is behind NAT, sending keep alives
Jun 16 16:14:11 Strongswan charon: 07[IKE] remote host is behind NAT



Jun 16 16:24:29 Strongswan charon: 13[ENC] parsed IKE_AUTH request 6 [ EAP/RES/TLS ]
Jun 16 16:24:29 Strongswan charon: 13[TLS] received TLS close notify
Jun 16 16:24:29 Strongswan charon: 13[TLS] sending TLS close notify

ipsec.conf:
```
config setup
    charondebug="ike 1, knl 1, cfg 1"
    uniqueids=no
conn azureTunnel

    auto=add
    keyexchange=ikev2
    fragmentation=yes
    forceencaps=yes
    dpdaction=clear
    dpddelay=300s
    ike=chacha20poly1305-sha512-curve25519-prfsha512,aes256gcm16-sha384-prfsha384-ecp384,aes256-sha1-modp1024
    esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1
    rekey=no

   
    left=vm_private_ip
    #leftid=This is the "Remote ID" string that macOS and iOS clients use.
    leftid=vm_public_ip
    #leftsubnets=the virtual network subnet
    leftsubnet=10.0.2.0/24
    #leftcert= the certificate used by the server
    leftcert=host-vpn-cert-publicip-v2.pem
    leftsendcert=always
    authby=pubkey
    #right=The %any option for the right side of the connection instructs the server to accept incoming connections from any remote client.
    right=%any
    #rightdns=These IP addresses are Google’s public DNS resolvers.
    rightdns=8.8.8.8,8.8.4.4
    #rightauth=This activates user authentication on the client. 
    rightauth=eap-tls
    #rightsourceip=This option instructs the server to assign private IP addresses to clients from the specified a  pool of IPs.
    rightsourceip=10.2.0.0/24
    #rightsubnet=One or multiple subnet can be defined, for each subnet a individual tunnel will be created
    rightsubnet=10.1.0.0/24

    type=tunnel
    eap_identity=%identity
```

on the client side:
```
`2023-06-16 19:15:20.577671 (pid/25733/com.apple.MTLCompilerService) <Notice>: created the implicit endpoint upfront

2023-06-16 19:15:20.577680 (pid/25733/com.apple.MTLCompilerService) <Notice>: draining messages from com.apple.MTLCompilerService

2023-06-16 19:15:20.578513 (pid/25692/com.apple.NetworkExtension.IKEv2Provider [25733]) <Notice>: extension already interrupted clean exit attempt

2023-06-16 19:15:20.611765 (user/501/com.apple.security.XPCKeychainSandboxCheck) <Notice>: internal event: WILL_SPAWN, code = 0

2023-06-16 19:15:20.611773 (user/501/com.apple.security.XPCKeychainSandboxCheck) <Notice>: service state: spawn scheduled

2023-06-16 19:15:20.611775 (user/501/com.apple.security.XPCKeychainSandboxCheck) <Notice>: service state: spawning

2023-06-16 19:15:20.611787 (user/501/com.apple.security.XPCKeychainSandboxCheck) <Notice>: launching: ipc (mach)

2023-06-16 19:15:20.612439 (user/501/com.apple.security.XPCKeychainSandboxCheck [25734]) <Notice>: xpcproxy spawned with pid 25734

2023-06-16 19:15:20.612453 (user/501/com.apple.security.XPCKeychainSandboxCheck [25734]) <Notice>: internal event: SPAWNED, code = 0

2023-06-16 19:15:20.612455 (user/501/com.apple.security.XPCKeychainSandboxCheck [25734]) <Notice>: service state: xpcproxy

2023-06-16 19:15:20.612462 (user/501/com.apple.security.XPCKeychainSandboxCheck [25734]) <Notice>: internal event: SOURCE_ATTACH, code = 0

2023-06-16 19:15:20.621792 (user/501/com.apple.security.XPCKeychainSandboxCheck [25734]) <Notice>: service state: running

2023-06-16 19:15:20.621801 (user/501/com.apple.security.XPCKeychainSandboxCheck [25734]) <Notice>: internal event: INIT, code = 0

2023-06-16 19:15:20.621838 (user/501/com.apple.security.XPCKeychainSandboxCheck [25734]) <Notice>: Successfully spawned XPCKeychainSandboxCheck[25734] because ipc (mach)

2023-06-16 19:15:20.625827 (pid/25734 [XPCKeychainSand]) <Notice>: uncorking exec source upfront

2023-06-16 19:15:20.625844 (pid/25734 [XPCKeychainSand]) <Notice>: created

2023-06-16 19:15:20.629601 (pid/25734 [XPCKeychainSand]) <Notice>: uncorking domain

2023-06-16 19:15:20.629931 (pid/25734 [XPCKeychainSand]) <Notice>: Service stub created for com.apple.foundation.UserScriptService

2023-06-16 19:15:20.630164 (pid/25734 [XPCKeychainSand]) <Notice>: Service stub created for com.apple.security.XPCAcmeService

2023-06-16 19:15:20.630164 (pid/25734 [XPCKeychainSand]) <Notice>: Service stub created for com.apple.security.XPCAcmeService

2023-06-16 19:15:20.918960 (pid/25692/com.apple.NetworkExtension.IKEv2Provider [25733]) <Notice>: exited due to SIGKILL | sent by nesessionmanager[437]

2023-06-16 19:15:20.918967 (pid/25692/com.apple.NetworkExtension.IKEv2Provider [25733]) <Notice>: process exited in a dirty state

2023-06-16 19:15:20.918972 (pid/25692/com.apple.NetworkExtension.IKEv2Provider [25733]) <Notice>: service state: exited

2023-06-16 19:15:20.918987 (pid/25692/com.apple.NetworkExtension.IKEv2Provider [25733]) <Notice>: internal event: EXITED, code = 0

2023-06-16 19:15:20.918992 (pid/25692/com.apple.NetworkExtension.IKEv2Provider [25733]) <Notice>: job state = exited

2023-06-16 19:15:20.919025 (pid/25692 [neagent]) <Notice>: service inactive: com.apple.NetworkExtension.IKEv2Provider

2023-06-16 19:15:20.919030 (pid/25692/com.apple.NetworkExtension.IKEv2Provider [25733]) <Notice>: service state: not running

2023-06-16 19:15:20.919038 (pid/25733 [NEIKEv2Provider]) <Notice>: shutting down`
```

**Reply by BR41N14C** [reply]
not sure if this helps:
```
List of X.509 End Entity Certificates

  subject:  "CN=51.11.22.34"
  issuer:   "CN=VPN ROOT CA"
  validity:  not before Jun 17 22:28:33 2023, ok
             not after  Jun 15 22:28:33 2028, ok (expires in 1824 days)
  serial:    1b:a2:18:37:a7:4c:81:4d
  altNames:  51.11.22.34, 51.11.22.34
  flags:     serverAuth ikeIntermediate 
  authkeyId: 11:25:0b:92:b9:2a:7f:ea:dc:b3:62:01:71:97:71:55:5c:48:97:9b
  subjkeyId: df:e4:7e:71:8e:f8:0b:e1:17:55:17:86:d9:8a:0a:45:9c:f3:62:70
  pubkey:    RSA 4096 bits, has private key
  keyid:     2e:f7:fb:96:ba:40:50:ef:b7:14:41:8f:d6:88:17:29:9f:b7:51:32
  subjkey:   df:e4:7e:71:8e:f8:0b:e1:17:55:17:86:d9:8a:0a:45:9c:f3:62:70

List of X.509 CA Certificates

  subject:  "CN=VPN ROOT CA"
  issuer:   "CN=VPN ROOT CA"
  validity:  not before Jun 17 22:21:59 2023, ok
             not after  Jun 14 22:21:59 2033, ok (expires in 3649 days)
  serial:    39:ce:aa:6c:f6:59:f1:a9
  flags:     CA CRLSign self-signed 
  subjkeyId: 11:25:0b:92:b9:2a:7f:ea:dc:b3:62:01:71:97:71:55:5c:48:97:9b
  pubkey:    RSA 4096 bits
  keyid:     67:69:66:e1:a9:69:ce:39:d9:eb:5f:62:28:01:40:43:60:0e:12:f7
```

**Reply by BR41N14C** [reply]
@tobiasbrunner 
checking the syslog, i can see i am getting logs generated by both ipsec and charon:
```
Jun 17 23:00:47 VPN-Strongswan ipsec[761]: 09[ENC] generating IKE_AUTH response 5 [ EAP/REQ/TLS ]
Jun 17 23:00:47 VPN-Strongswan ipsec[761]: 09[NET] sending packet: from 10.0.2.9[4500] to client_pub_ip[4500] (96 bytes)
Jun 17 23:00:47 VPN-Strongswan ipsec[761]: 10[NET] received packet: from client_pub_ip[4500] to 10.0.2.9[4500] (96 bytes)
Jun 17 23:00:47 VPN-Strongswan charon: 10[TLS] sending TLS close notify
Jun 17 23:00:47 VPN-Strongswan ipsec[761]: 10[ENC] parsed IKE_AUTH request 6 [ EAP/RES/TLS ]
Jun 17 23:00:47 VPN-Strongswan ipsec[761]: 10[TLS] received TLS close notify
Jun 17 23:00:47 VPN-Strongswan charon: 10[ENC] generating IKE_AUTH response 6 [ EAP/REQ/TLS ]
Jun 17 23:00:47 VPN-Strongswan charon: 10[NET] sending packet: from 10.0.2.9[4500] to client_pub_ip[4500] (96 bytes)
```

**Reply by BR41N14C** [reply]
starting to think this is something to do with Azure, vm is in a virtual network with a network gateway

**Reply by tobiasbrunner** [reply]
While there are issues with Azure and EAP-TLS, that's only the case when used via RADIUS (see e.g. https://github.com/strongswan/strongswan/discussions/1658#discussioncomment-5684577). In IKEv2 messages it can't modify the certificates.

The client log only shows cleanup/shutdown stuff after the fact. There have to be more messages before that that show the initiation of the connection and possibly the error condition that lead to the shutdown.
